lib/adapters/http/read.js

```diff
@@ -144,26 +180,81 @@ class ReadHTTP extends AdapterRead {
     }
 
     read(from, to, limit, iterator) {
-        // The first read call should kick off the actual fetch request.
-        //
-        // adapter returns a promise that either rejects with an error or resolves
-        // with an iterator object  that handles paging through the results in
-        // chunks that we return as the read state.
-        if (! iterator) {
-            return this.fetch(from, to)
-            .then((iterator) => {
-                return iterator.next(to);
-            });
-        } else {
-            return iterator.next(to);
+        // The first read call should kick off the actual fetch request. This
+        // returns a promise that either rejects with an error or resolves with
+        // an iterator object that handles paging through the results in chunks
+        // that we return as the read state.
+        var url = this.url;
+
+        if (this.link) {
+            url = this.link;
         }
+
+        var getIterator = iterator
+            ? Promise.resolve(iterator)
+            : this.fetch(url, from, to);
+
+        return getIterator
+        .then((iterator) => {
+            return iterator.next();
+        })
+        .then((result) => {
+            // To support paging, check if the current iterator (which is
+            // returned in result.state) got any points for the current page. If
+            // so, then advance the page and keep going. Otherwise just stop.
+            if (this.pageParam && result.readEnd) {
+                if (result.state.total === 0) {
+                    this.logger.debug('no results for page', this.pageCount, ' -- paging completed');
+                } else {
+                    this.logger.debug('got', result.state.total, 'results for page', this.nextPage);
+
+                    switch(this.pageUnit) {
+                        case 'request':
+                            this.pageCount++;
+                            break;
+
+                        case 'record':
+                            this.pageCount += result.state.total;
+                    }
+
+                    result.readEnd = undefined;
+                    result.state = undefined;
+                }
+            }
+            
+            // if followLinkHeader is set and there is an actual linkHeader
+            // returned from the last fetch then lets follow that link
+            if (this.followLinkHeader && result.state.linkHeader) {
+                this.link = result.state.linkHeader;
+                result.readEnd = undefined;
+                result.state = undefined;
+            }
+
+            return result;
+        });
     }
 
-    fetch(from, to) {
+    fetch(thisURL, from, to) {
```

Comment:
url is the lib required up above `var url = require('url')` and so I wanted to avoid the name collision and what I'll do instead is simply `var urlParse = require('url').parse` and be able to use the url var here just fine.